ros/src/tl_detector/light_classification/tl_classifier.py

```diff
@@ -85,33 +93,110 @@ def get_classification(self, image):
             int: ID of traffic light color (specified in styx_msgs/TrafficLight)
 
         """
-        t0 = rospy.Time.now()
-        image = cv2.resize(image, (self.resize_width, self.resize_height))
-        np_image_data = np.asarray(image)
-        np_final = np.expand_dims(np_image_data, axis=0)
-        np_final = resnet50.preprocess_input(np_final.astype('float64'))
-
-        t1 = rospy.Time.now()
-        dt1 = t1 - t0
-
-        yhat = self.sess.run(self.y, feed_dict={self.x: np_final})
-
-        dt2 = rospy.Time.now() - t1
-
-        yhat = yhat[0]
-        y_class = yhat.argmax(axis=-1)
         labels = self.labels
 
-        rospy.loginfo('%s (%.2f%%) : ImagePrep time (s) : %f GPU time (s) : %f', labels[y_class],
-                      yhat[y_class]*100, dt1.to_sec(), dt2.to_sec())
+        t0 = rospy.Time.now()
+
+        # Switch classification method between site test (classify by multiple
+        # image clips) vs simulator (classify by single full image)
+        if self.use_image_clips is True:
+            # Classify by multiple image clips
+            ###chop image up
+            detect = False
+            # left corner x co-ords to split 800 pixels into 5 sections of 224 pixels
+            startx = [0,152,288,424,576]
```

Comment:
Since this has some hard-coded dimensions based on 800x600 input image size from the data, maybe we should add an initial image resize image = cv2.resize(image, (800, 600)) to guarantee the correct starting dimension just in case the camera input size changes?

Reply:
Sounds reasonable and a good precaution as raw image size was not 800x600 at start.